android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -878,7 +942,7 @@ private MediaSource buildMediaSource(Uri uri, String overrideExtension, DrmSessi
         if (uri == null) {
             throw new IllegalStateException("Invalid video uri");
         }
-        int type = Util.inferContentType(!TextUtils.isEmpty(overrideExtension) ? "." + overrideExtension
+        int type = Util.inferContentType(!TextUtils.isEmpty(overrideExtension) && !overrideExtension.equals(DOWNLOADED_CONTENT) ? "." + overrideExtension
```

Comment:
Magic strings don't make very good APIs, should this be an explicit boolean or enum property?

Reply:
`extension` is a string type in `react-native-video`. We currently use it to distinguish between downloaded videos and online streams. I don't think it's worth spending time changing its type.

Reply:
I was thinking more of an explicit setDownload(boolean), rather than trying to hack the download indicator into the extension.

Reply:
The 'download' string is passed from the JavaScript side, and it is also used on the iOS side. In this PR, I aim to maintain the functionality of the old approach without refactoring it.